pkg/clusteragent/admission/mutate/autoinstrumentation/target_mutator.go

```diff
@@ -311,41 +311,62 @@ type targetInternal struct {
 
 // getTarget determines which target to use for a given a pod, which includes the set of tracing libraries to inject.
 func (m *TargetMutator) getTarget(pod *corev1.Pod) *targetInternal {
-	if target := m.getTargetFromAnnotation(pod); target != nil {
-		return target
+	result := m.getTargetFromAnnotation(pod)
+	if !result.shouldContinue {
+		return result.target
 	}
 
 	return m.getMatchingTarget(pod)
 }
 
+type annotationResult struct {
+	shouldContinue bool
+	target         *targetInternal
+}
+
 // getTargetFromAnnotation determines which tracing libraries to use given
-func (m *TargetMutator) getTargetFromAnnotation(pod *corev1.Pod) *targetInternal {
+func (m *TargetMutator) getTargetFromAnnotation(pod *corev1.Pod) *annotationResult {
```

Comment:
Nit:
Leaving this as a pointer return while not checking it in the caller leaves room for a future null-pointer condition.

We could just return `annotationResult` rather than `*annotationResult`